app/routes.py

```diff
@@ -1,2 +1,253 @@
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": False    
+        }}
+    
+    
+    elif request.method == "PUT":
+        data = request.get_json()
+        
+        task.title = data["title"]
+        task.description = data["description"]
+        task.completed_at = data["completed_at"]
+
+        db.session.commit()
+        
+        data_response= {
+        "task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": is_complete(task.completed_at)
+        }
+    }
+        return make_response(jsonify(data_response),200)
+   
+
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        
+        delete_id_response= f'Task {task.task_id} "{task.title}" successfully deleted'
+        
+        return make_response(jsonify({"details":delete_id_response}))
+
+
+@tasks_bp.route("/<task_id>/mark_complete", methods = ["PATCH"])
+def handle_complete(task_id):
+    task = Task.query.get(task_id)
+    
+    if task is None :
+        return make_response("", 404)
+
+    task.completed_at = datetime.now()
+
+    db.session.commit()
+    
+    slack_url='https://slack.com/api/chat.postMessage'
+    token= os.environ.get("SLACK_TOKEN")
+    params = {
+        'channel': 'task-notifications',
+        'text': f'Someone just completed the task {task.title}'
+    }
+    headers={
+        'Content-type': 'application/json',
+        'Authorization': f"Bearer {token}"
+    }
+
+    requests.post(slack_url,json=params,headers=headers)
+    data_response = {
+        "task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": True 
+        }
+    }
+
+
+    return make_response(jsonify(data_response),200)
+  
+
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods = ["PATCH"])
+def handle_incomplete(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return make_response("",404)
+    
+    data_response = {
+        "task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": False
+        }
+    }
+    db.session.commit()
+    if task.completed_at is None : 
+        return make_response(jsonify(data_response),200)
+    
+    task.completed_at = None
+    return make_response(jsonify(data_response),200)
+
+
+
+
+# Routes for Goal
+
+@goals_bp.route("", methods=["GET","POST"])
+def handle_goals():
+    goal_reponse=[]
+    goals = Goal.query.all()
+  
+    if request.method == "GET":
+    
+        for goal in goals:
+            
+            goal_reponse.append({
+                "id": goal.goal_id,
+                "title":goal.title,
+            })
+        return make_response(jsonify(goal_reponse),200)
+
+    else:
```

Comment:
what to make sure `request.mthod == 'POST` here
```suggestion
    elif request.method == 'POST':
```